File: benchmark/src/main/java/com/jonathansteele/parsnip/parnsip/small/ParsnipSmallXmlBenchmark.java
package com.jonathansteele.parsnip.parnsip.small;

import com.jonathansteele.parsnip.Parsnip;
import com.jonathansteele.parsnip.XmlAdapter;

public class ParsnipSmallXmlBenchmark {
    public void parse(String xml) throws Exception {
        Parsnip parsnip = new Parsnip.Builder().build();
        XmlAdapter<Employee> employeeAdapter = parsnip.adapter(Employee.class);
        Employee employee = employeeAdapter.fromXml(xml);
        System.out.println(getClass().getSimpleName() + " " + employee.name);
    }
}

File: benchmark/src/main/kotlin/com/jonathansteele/parsnip/parsnip/small/ParsnipSmallXmlBenchmark.kt
package com.jonathansteele.parsnip.parsnip.small

import com.jonathansteele.parsnip.Parsnip

fun parseParsnipXml(xml: String?) {
    val parsnip = Parsnip.Builder().build()
    val employeeAdapter = parsnip.adapter(
        Employee::class.java
    )
    val employee = employeeAdapter.fromXml(xml!!)
    println("Parsnip " + employee.name)
}